### DISCUSSION TITLE
First Complete Draft

### DISCUSSION THREAD

**Discussion by ckirsch** [ckirsch/book#7] [Announcements]
This is the first complete draft of the book "Elementary Computer Science: From Bits and Bytes to the Universality of Computing" by Christoph Kirsch.

<hr /><em>This discussion was created from the release <a href='https://github.com/ckirsch/book/releases/tag/'>First Complete Draft</a>.</em>

**Comment by zicklag** [comment]
Thanks for this book! I just discovered it yesterday, I've only gotten to page 47, but I love the idea, and your general approach and attitude.

I'm already a very competent programmer, experienced with Rust, Typescript, and more, but as a learning exercise I wanted to learn how to make a self-contained OS that can compile itself. That led me to small C compilers such as [Chibicc](https://github.com/rui314/chibicc) and Selfie.

I love how with Selfie you restrict everything to such a simple, but useful, base, and how self contained it all is, and I'm excited to learn more about how it works and hopefully take some of the concepts with me into my own project. It's fun to actually get back to the "dawn of computers" so to speak and consider what it's like to do useful computing without having to depend on millions of lines of code from other people's projects. 😃
